presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestRoles.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.tests.hive;
+
+import com.google.common.collect.ImmutableSet;
+import com.teradata.tempto.AfterTestWithContext;
+import com.teradata.tempto.BeforeTestWithContext;
+import com.teradata.tempto.ProductTest;
+import com.teradata.tempto.assertions.QueryAssert;
+import com.teradata.tempto.query.QueryExecutor;
+import com.teradata.tempto.query.QueryResult;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.facebook.presto.tests.TestGroups.AUTHORIZATION;
+import static com.facebook.presto.tests.TestGroups.HIVE_CONNECTOR;
+import static com.facebook.presto.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static com.facebook.presto.tests.TestGroups.ROLES;
+import static com.facebook.presto.tests.utils.QueryExecutors.connectToPresto;
+import static com.facebook.presto.tests.utils.QueryExecutors.onHive;
+import static com.facebook.presto.tests.utils.QueryExecutors.onPresto;
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toSet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestRoles
+        extends ProductTest
+{
+    private static final String ROLE1 = "role1";
+    private static final String ROLE2 = "role2";
+    private static final String ROLE3 = "role3";
+    private static final Set<String> TEST_ROLES = ImmutableSet.of(ROLE1, ROLE2, ROLE3);
+
+    @BeforeTestWithContext
+    public void setUp()
+            throws Exception
+    {
+        onHive().executeQuery("SET ROLE admin");
+        cleanup();
+    }
+
+    @AfterTestWithContext
+    public void tearDown()
+            throws Exception
+    {
+        cleanup();
+    }
+
+    public void cleanup()
+    {
+        Set<String> existentRoles = listRoles();
```

Comment:
This seems kinda aggressive. I think we should only remove roles which are used in this test. That is `ROLE{1,3}`.
And then when we check user roles only take into account the intersection of roles returned by metastore and `TEST_ROLES`.